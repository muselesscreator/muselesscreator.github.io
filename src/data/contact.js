import { StrictDict } from '@muselesscreator/strict-dict';
import avatar from '../assets/ampersand.png';

const contactInfo = ({
  name: 'Ben Warzeski',
  email: 'bwarzeski@gmail.com',
  title: 'Full-Stack Software Engineer',
  phone: '+1 (941) 387-5092',
  phoneRaw: '19413875092',
  github: 'https://www.github.com/muselesscreator',
  linkedIn: 'https://www.linkedin.com/in/benjamin-warzeski-69111518/',
  avatar,
});

export default contactInfo;
