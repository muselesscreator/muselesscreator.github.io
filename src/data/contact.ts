import { StrictDict } from '@muselesscreator/strict-dict';
import avatar from '../assets/ampersand.png';

export type ContactInfo = {
  name: string;
  email: string;
  title: string;
  phone: string;
  phoneRaw: string;
  github: string;
  linkedIn: string;
  avatar: string;
};

const contactInfo = StrictDict({
  name: 'Ben Warzeski',
  email: 'bwarzesk@gmail.com',
  title: 'Full-Stack Software Engineer',
  phone: '+1 (941) 387-5092',
  phoneRaw: '19413875092',
  github: 'https://www.github.com/muselesscreator',
  linkedIn: 'https://www.linkedin.com/in/benjamin-warzeski-69111518/',
  avatar,
}) as ContactInfo;

export default contactInfo;
