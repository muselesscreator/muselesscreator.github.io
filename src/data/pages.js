import { StrictDict } from '@muselesscreator/strict-dict';

export const pages = StrictDict({
  about: 'about',
  contact: 'contact',
  resume: 'resume',
  portfolio: 'portfolio',
  blog: 'blog',
});

export const pageNames = StrictDict({
  [pages.about]: 'About',
  [pages.contact]: 'Contact',
  [pages.resume]: 'Resume',
  [pages.portfolio]: 'Portfolio',
  [pages.blog]: 'Blog',
});

export default pages;
