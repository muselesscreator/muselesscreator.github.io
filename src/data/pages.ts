import { StrictDict } from '@muselesscreator/strict-dict';

export const pages = StrictDict({
  about: 'about',
  contact: 'contact',
  resume: 'resume',
  portfolio: 'portfolio',
  demo: 'demo',
  blog: 'blog',
}) as Record<string, string>;

export const pageNames = StrictDict({
  [pages.about]: 'About',
  [pages.contact]: 'Contact',
  [pages.resume]: 'Resume',
  [pages.portfolio]: 'Portfolio',
  [pages.blog]: 'Blog',
  [pages.demo]: 'Demos',
}) as Record<string, string>;

export default pages;
