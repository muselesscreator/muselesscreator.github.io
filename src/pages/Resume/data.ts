import 'react';
import { StrictDict } from '@muselesscreator/strict-dict';

export type Field = ({
  type: 'text',
  text: string,
} | {
  type: 'list',
  title?: string,
  items: Field[],
});

export type ResumeSectionType = {
  items: {
    date: string,
    title: string,
    fields: Field[],
  }[],
};

type Resume = {
  resume: {
    education: ResumeSectionType,
    experience: ResumeSectionType,
  },
};


const pageData = StrictDict({
  resume: {
    education: {
      items: [
        {
          title: 'Tulane University',
          date: '2006 — 2011',
          fields: [
            { type: 'text', text: 'Bachelor of Science in Mathematics with minors in Music and "Music, Science and Technology".' },
          ],
        },
      ],
    },
    experience: {
      items: [
        {
          title: 'Senior Software Engineer II, UI-Lead - EdX/2U Inc, Boston, MA',
          date: '2020 - 2024',
          fields: [
            { type: 'text', text: 'I served as the Frontend technical leadership for my squad across formation and delivery of multiple projects, as well as serving as the Project Manager for later projects, including the Learner Home and ORA redesign projects.' },
            { type: 'text', text: 'My core priorities and contributions outside of project delivery came in the form of testing, community, and training.' },
            { type: 'list', title: 'Community-Building', items: [
              { type: 'text', text: 'Organized and managed the Fedx (frontend-at-edx) community, integrating rotating leadership, a manage-able charter, engineering leadership support/sponsorship, and well-defined responsibilities.' },
              { type: 'text', text: 'This community serves as the hub for frontend development across teams at a company with no formal frontend organization.  This provided us a place and avenue for discussing/resolving tricky “glue” type problems that affected many teams, but none specifically enough to get funding otherwise (dependency upgrades, pattern updates, adoption of new technology, definition and consolidation of coding standards).'},
            ] },
            { type: 'list', title: 'Training', items: [
              { type: 'text', text: 'In addition to training up a junior frontend dev on my team as well as training my backend devs to be comfortable working quickly and effectively in our React code, I spent a great deal of time training up developers on other teams on frontend development and testing techniques.' },
              { type: 'text', text: 'Organized and managed the frontend study group, and integrated it with the fedx community.  I wrote and ran a variety of training sessions on testing, hooks, react development, internal tools, typescript, and more.' },
            ] },
            { type: 'list', title: 'Testing', items: [
              { type: 'text', text: 'I maintain a high standard of testing in my projects, with a tendency to release with close to 100% test coverage.  I have a focus on accurate/easy unit tests that provide maximum value for minimum creation/maintenance time.' },
              { type: 'text', text: 'As part of my testing pattern, I have developed a testing library to serve as a replacement for Enzyme, which is getting deprecated in future versions of React, which has begun being used by members of the community to mass-overhaul testing patterns.' },
            ] },
          ],
        }, {
          title: 'Senior Software Engineer - ReadCoor Inc, Boston, MA',
          date: '2018 - 2020',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'Facilitated technology comparison process across teams to determine in which new UI technology the disparate  User Interfaces at the company should work towards being re-written, managing to acquire broad buy-in  across multiple development teams with very different focuses and existing JS workflows.  • Succeeded in pushing to implement a stable Javascript development workflow, with documented coding  guidelines, UI-generation scripts, reusable plug-in JS components and modules, and a shared theming.' },
              { type: 'text', text: 'Lead re-write of multiple existing user interfaces into React/Redux.' },
              { type: 'text', text: 'Lead the requirements generation and implementation of the Customer Device Interface, providing control  over all the parts of the genetic sequencing scan setup and run procedure, with input from and co-ordination  with UX designer. As part of this process, I helped lead the generation of initial design and requirements,  provided 90% of JS implementation, facilitated/lead backend API updates/integration, and generated a full  set of requirements in Testlink along with a development manual test plan for engineer testers.' },
            ] },
          ],
        }, {
          title: 'Software Engineer, UI Lead - Optimus Ride, Boston, MA',
          date: '2017 — 2018',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'As Lead UI Developer, implemented web-based customer-facing User Interfaces in React/Redux, targeted at  many levels of customer, including rider, dispatchers, vehicle operators, and fleet managers.' },
              { type: 'text', text: 'As Scrum Master for Fleet Software team: designed and enforced an agile system to bring structure to a  disorganized team. Implemented 2-week sprints, with regular agile meetings, resulting in improved output,  and vastly improved targeting of priorities by enforcing a structure around requirements-gathering and  problem definition from stakeholders.' },
              { type: 'text', text: 'As SW Engineer, built python backend services in support of web user interfaces, allowing tracking of a  meta-contextual state across system components and tracking/controlling the data driving the user  experience workflow throughout and across all user levels/types.' },
            ]}
          ],
        }, {
          title: 'Software Engineer, Software Engineer II - Bio-Rad, Boston MA',
          date: '2014 — 2017',
          fields: [
            { type: 'list', items: [
              { type: 'text', text: 'As Project Manager for the Customer User Interface: managed coordination between manufacturing,  marketing, biology, and software departments, along with an external UX contracting firm, to produce a  touch-screen user interface for the MonoRAyL system.' },
              { type: 'text', text: 'As Lead Developer for the Customer User Interface: produced a touch-screen web app based around a  Python web server, responsible for guiding the user through the steps of an experiment, while also managing  the hardware interaction, data access, user management, and software/firmware updates.' },
              { type: 'text', text: 'As Lead Developer for the Internal Development User Interface: produced a complex, robust one-page app  providing control over all aspects of the complex biomedical microfluidic device:' },
              { type: 'list', items: [
                { type: 'text', text: 'Provided controls and status readouts for over 40 physical sensors and interfaces, including: electrodes,  cameras, stepper motors, thermal plates, pressure pumps/valves, lasers, and LEDs' },
                { type: 'text', text: 'Provided sub-tools for manufacturing and software teams allowed management of all running sub processes and internal flags for the system as well as interactive tools for running end-of-line testing and  managing machine configurations.' },
                { type: 'text', text: 'Developed and maintained tools and APIs for inter-process communication, database management and interaction, unit testing, asynchronous process coordination, and hardware interactions.' },
              ]},
            ]},
          ],
        }, {
          title: 'Co-Founder/Lead Developer - Nerd Pile Gaming, Boston, MA',
          date: '2013 — 2016',
          fields: [
            { type: 'text', text: 'For the studio’s first iOS/Android release, Farmageddon' },
            { type: 'list', items: [
              { type: 'text', text: 'Lead Developer, implementing core gameplay APIs in Lua using CoronaSDK.' },
              { type: 'text', text: 'Lead Sound Engineer, producing sound effects, voice overs and seven music compositions.' },
            ] },
            { type: 'text', text: 'Post-Farmageddon' },
            { type: 'list', items: [
              { type: 'text', text: 'Lead Developer for further projects in Unity' },
              { type: 'text', text: 'Developed 2D and 3D interfaces for a variety of puzzle games' },
              { type: 'text', text: 'Lead Sound Engineer, four music compositions for PuzzlePlanets' },
            ] },
          ],
        },
      ],
    },
  },
});

export default pageData as Resume;
