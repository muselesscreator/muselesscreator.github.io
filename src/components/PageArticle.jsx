import 'react';
import About from '../pages/About';
import Blog from '../pages/Blog';
import Contact from '../pages/Contact';
import Portfolio from '../pages/Portfolio';
import Resume from '../pages/Resume';
import { pages, pageNames } from '../data/pages';

const articleComponents = {
  [pages.about]: About,
  [pages.blog]: Blog,
  [pages.contact]: Contact,
  [pages.portfolio]: Portfolio,
  [pages.resume]: Resume,
};

const PageArticle = ({ activePage }) => {
  const ArticleComponent = articleComponents[activePage];
  return (<ArticleComponent />);
};

export default PageArticle;
