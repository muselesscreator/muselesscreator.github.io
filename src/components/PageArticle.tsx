import 'react';
import About from '../pages/About';
import Contact from '../pages/Contact';
import Portfolio from '../pages/Portfolio';
import Resume from '../pages/Resume';
import Demos from '../pages/Demos';
import { pages } from '../data/pages';

const articleComponents = {
  [pages.about]: About,
  [pages.contact]: Contact,
  [pages.portfolio]: Portfolio,
  [pages.resume]: Resume,
  [pages.demo]: Demos,
};

const PageArticle = ({ activePage }: { activePage: string }) => {
  const ArticleComponent = articleComponents[activePage];
  return (<ArticleComponent />);
};

export default PageArticle;
