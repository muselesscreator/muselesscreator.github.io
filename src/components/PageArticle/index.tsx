import About from '~/pages/About';
import Contact from '~/pages/Contact';
import Portfolio from '~/pages/Portfolio';
import Resume from '~/pages/Resume';
import Demos from '~/pages/Demos';
import APIDocs from '~/pages/APIDocs';
import { pages } from '~/data/pages';

export const articleComponents = {
  [pages.about]: About,
  [pages.contact]: Contact,
  [pages.portfolio]: Portfolio,
  [pages.resume]: Resume,
  [pages.demo]: Demos,
  [pages.api]: APIDocs,
};

const PageArticle = ({ activePage }: { activePage: string }) => {
  const ArticleComponent = articleComponents[activePage];
  return (<ArticleComponent />);
};

export default PageArticle;
