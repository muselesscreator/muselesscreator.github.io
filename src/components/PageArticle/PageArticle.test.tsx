import {
  vi,
  test,
  describe,
  expect,
} from 'vitest';
import { pages } from '~/data/pages';

import { shallow } from '@muselesscreator/react-shallow-snapshot';

import PageArticle, { articleComponents } from '.';

vi.mock('~/pages/About', () => ({ default: 'About' }));
vi.mock('~/pages/Contact', () => ({ default: 'Contact' }));
vi.mock('~/pages/Portfolio', () => ({ default: 'Portfolio' }));
vi.mock('~/pages/Resume', () => ({ default: 'Resume' }));
vi.mock('~/pages/Demos', () => ({ default: 'Demos' }));

describe('PageArticle component', () => {
  test('should render About page if activePage is about', () => {
    const el = shallow(<PageArticle activePage="about" />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance).toEqual(shallow(<articleComponents.about />).instance);
  });
  test('should render Contact page if activePage is contact', () => {
    const el = shallow(<PageArticle activePage="contact" />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance).toEqual(shallow(<articleComponents.contact />).instance);
  });
  test('should render Portfolio page if activePage is portfolio', () => {
    const el = shallow(<PageArticle activePage="portfolio" />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance).toEqual(shallow(<articleComponents.portfolio />).instance);
  });
  test('should render Resume page if activePage is resume', () => {
    const el = shallow(<PageArticle activePage="resume" />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance).toEqual(shallow(<articleComponents.resume />).instance);
  });
  test('should render Demos page if activePage is demos', () => {
    const el = shallow(<PageArticle activePage={pages.demo} />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance).toEqual(shallow(<articleComponents.demo />).instance);
  });
});

