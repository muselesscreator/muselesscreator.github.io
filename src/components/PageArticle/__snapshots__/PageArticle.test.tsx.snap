// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageArticle component > should render About page if activePage is about 1`] = `"<About />"`;

exports[`PageArticle component > should render Contact page if activePage is contact 1`] = `"<Contact />"`;

exports[`PageArticle component > should render Demos page if activePage is demos 1`] = `"<Demos />"`;

exports[`PageArticle component > should render Portfolio page if activePage is portfolio 1`] = `"<Portfolio />"`;

exports[`PageArticle component > should render Resume page if activePage is resume 1`] = `"<Resume />"`;
